libgadget/density.c

```diff
@@ -528,7 +551,13 @@ void density_check_neighbours (int i, TreeWalk * tw) {
             {
                 if(P[i].Type == 0 && fabs(P[i].NumNgb - desnumngb) < 0.5 * desnumngb)
                 {
-                    double fac = 1 - (P[i].NumNgb - desnumngb) / (NUMDIMS * P[i].NumNgb) * SPHP(i).DhsmlDensityFactor;
+                    MyFloat DensFac;
+                    if(DENSITY_GET_PRIV(tw)->DhsmlDensityFactor)
```

Comment:
Maybe you want to use the master flag (cleary shows the motivation of the branching ) instead of a derived condition.

Reply:
Yes, I do!